src/cc/usdt.cc

```diff
@@ -210,14 +229,44 @@ std::string Context::resolve_bin_path(const std::string &bin_path) {
   return result;
 }
 
-Probe *Context::find_probe(const std::string &probe_name) {
-  for (Probe *p : probes_) {
+std::shared_ptr<Probe> Context::get(const std::string &probe_name) {
```

Comment:
As a personal choice I tend to avoid shared_ptr, partly due to bad history and partly due to design philosophy. I haven't read through this code in enough depth to say whether it can be done differently, so I'll just ask: can it be done differently?